operations/helm/charts/mimir-distributed/templates/distributor/distributor-dep.yaml

```diff
@@ -8,7 +8,9 @@ metadata:
     {{- toYaml .Values.distributor.annotations | nindent 4 }}
   namespace: {{ .Release.Namespace | quote }}
 spec:
+  {{- if .Values.distributor.replicas }}
```

Comment:
can we make use of the `kindIs` function here? Only skip the replicas if their type is not an int and they are false-y? Something like `if and (kindIs "int" .Values.distributor.replicas) (.Values.distributor.replicas)`?

Reply:
So
```
  {{- if or (kindIs "float64" .Values.distributor.replicas) (.Values.distributor.replicas) }}
  replicas: {{ .Values.distributor.replicas }}
  {{- end }}
```
seems to be working fine replicas set to 0, or other number works, replicas set to "1" also works. And than setting replicas to null doesn't set it.

Should I then update the deploments with this option?

Reply:
strangely when I use `helm --set distributor.replicas=0` this doesn't work, but it works when i set `0` in the values files.


```bash
helm template . --set distributor.replicas=0 2>/dev/null | yq 'select(.metadata.name=="release-name-mimir-distributor" and .kind=="Deployment") | .spec.replicas'
null
```

Reply:
Did you know that when you use the values file to set a number it's `float64` but when you use the `--set` with the cli to set a number it's `int64`? 
I didn't. Helm.... 

Anyway added a check for wether it's float or int. Seem to work now as expected in all the cases:
* Set via values file
  * null
  * 1
  * 2
  * 0
* Set via --set
  * null
  * 1
  * 2 
  * 0

Reply:
that ended up being way more complicated than I thought 🙈 thanks for investigating it